app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -112,25 +112,50 @@ if [ x$ID = xrhel ] || [ x$ID = xcentos ]; then
 <%= "          --data 'remote_execution_interface=#{@remote_execution_interface}' \\\n" if @remote_execution_interface.present? -%>
 <%= "          --data 'packages=#{@packages}' \\\n" if @packages.present? -%>
 
-}
+    }
 
     <% if @force -%>
-    yum remove -y katello-ca-consumer*
+    yum remove -y katello-ca-consumer* subscription-manager
     <% end -%>
 
-    CONSUMER_RPM=$(mktemp --suffix .rpm)
-    curl --silent --show-error --output $CONSUMER_RPM <%= subscription_manager_configuration_url(hostname: @url_host) %>
+    KATELLO_SERVER_CA_CERT=/etc/rhsm/ca/<%= @url.host %>.pem
+    RHSM_CFG=/etc/rhsm/rhsm.conf
 
-    # Workaround for systems with enabled FIPS,
-    # where installation of RPM generated on RHEL7 cause 'no digest' error
-    # See https://projects.theforeman.org/issues/32068
-    if [ "$(cat /proc/sys/crypto/fips_enabled)" = "1" ]; then
-        rpm -ivh --nodigest --nofiledigest $CONSUMER_RPM
-    else
-        yum localinstall $CONSUMER_RPM -y
+    if ! command -v subscription-manager &> /dev/null
+    then
+      yum install -y subscription-manager
```

Comment:
I'd consider updating this to be safe for use with both `dnf` and `yum` for when the day comes they remove `yum`.